### PR TITLE
e2e: tag documentation + sorting

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Adding new entries at the bottom is tempting, but increases the risk of merge
conflicts between unrelated PRs. Let's use alphabetic order instead.

Adding a doc comment for all existing items makes it more obvious that new
items should be documented more carefully. It also has the welcome side effect
that each item gets indented independently from the others.

#### Which issue(s) this PR fixes:
Related-to:
- https://github.com/kubernetes/kubernetes/pull/123183#issuecomment-1941156881
- https://github.com/kubernetes/enhancements/tree/master/keps/sig-testing/3041-node-conformance-and-features

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/environment/environment.go [hunk 1]
```diff
@@ -23,8 +23,12 @@ import (
 )
 
 var (
+	// Please keep the list in alphabetical order.
+
 	// The test does not work in UserNS (for example, `open /proc/sys/kernel/shm_rmid_forced: permission denied`).
 	NotInUserNS = framework.WithEnvironment(framework.ValidEnvironments.Add("NotInUserNS"))
+
+	// Please keep the list in alphabetical order.
```
Comment (pohly) [inline test/e2e/environment/environment.go:31]: This comment at the bottom doesn't make a lot of sense here, but it is relevant for the larger lists: if it had been present earlier, then the diff in https://github.com/kubernetes/kubernetes/commit/6073d1cd3d58384d12a750bda749ff1922812be3#diff-337683b7f1560f35fe4a306d97241505a4b6e852d0aea485221d7f6ab67524d8 would have shown the request to keep the list sorted.

Adding that comment only at the top is not enough for that.

File: test/e2e/feature/feature.go [hunk 2]
```diff
@@ -23,116 +23,339 @@ import (
...
-	NetworkingPerformance                   = framework.WithFeature(framework.ValidFeatures.Add("Networking-Performance"))
-	NetworkPolicy                           = framework.WithFeature(framework.ValidFeatures.Add("NetworkPolicy"))
-	NodeAuthenticator                       = framework.WithFeature(framework.ValidFeatures.Add("NodeAuthenticator"))
-	NodeAuthorizer                          = framework.WithFeature(framework.ValidFeatures.Add("NodeAuthorizer"))
-	NodeLogQuery                            = framework.WithFeature(framework.ValidFeatures.Add("NodeLogQuery"))
-	NodeOutOfServiceVolumeDetach            = framework.WithFeature(framework.ValidFeatures.Add("NodeOutOfServiceVolumeDetach"))
-	NoSNAT                                  = framework.WithFeature(framework.ValidFeatures.Add("NoSNAT"))
+	// Please keep the list in alphabetical order.
+
+	// TODO: document the feature (owning SIG, when to use this feature for a test)
```
Comment (pohly) [inline test/e2e/feature/feature.go:28]: Documenting "when to use this feature" has two purposes:
- making it clear when to reuse an existing label for a new test
- together with the owning SIG, make it clear which jobs should test the feature and how it needs to be configured

Comment (pohly) [reply test/e2e/feature/feature.go:28]: The "how it needs to be configured" might have to be called out explicitly?

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Pull request merged**